components/tabs/src/sync/record.rs

```diff
@@ -122,14 +122,25 @@ fn parse_last_used(last_used_val: &JsonValue) -> Result<u64> {
     Ok(last_used)
 }
 
-fn parse_string_from_json(field_name: &str, val: Option<&JsonValue>) -> Result<String> {
-    Ok(val
-        .ok_or_else(|| serde_json::Error::custom(format!("missing field `{}`", field_name)))?
-        .as_str()
-        .ok_or_else(|| {
-            serde_json::Error::custom(format!("invalid `{}`, expected string", field_name))
-        })?
-        .to_string())
+fn parse_string_from_json(
+    field_name: &str,
+    val: Option<&JsonValue>,
+    can_be_empty: bool,
```

Comment:
Naming consistency nit: "empty" -> "missing" for consistency with the error message that you throw when this value is `false`.